mox/impl/src/lib.rs

```diff
@@ -1,245 +1,239 @@
-            SnaxItem::SelfClosingTag(t) => MoxItem::TagNoChildren(MoxTagNoChildren::from(t)),
-            SnaxItem::Fragment(SnaxFragment { children }) => {
-                MoxItem::Fragment(children.into_iter().map(MoxItem::from).collect())
-            }
-            SnaxItem::Content(atom) => {
-                MoxItem::Content { span: atom.span(), stream: wrap_content_tokens(atom) }
-            }
-        }
-    }
+struct MoxAttr {
+    name: syn::Ident,
+    value: Option<syn::Expr>,
 }
 
-fn wrap_content_tokens(tt: TokenTree) -> TokenStream {
-    let mut new_stream = quote!(#tt);
-    match tt {
-        TokenTree::Group(g) => {
-            let mut tokens = g.stream().into_iter();
-            if let Some(TokenTree::Punct(p)) = tokens.next() {
-                if p.as_char() == '%' {
-                    // strip the percent sign off the front
-                    new_stream = TokenStream::new();
-                    new_stream.extend(tokens);
-                    new_stream = quote!(text(format!(#new_stream)));
-                }
-            }
-        }
-        tt @ TokenTree::Ident(_) | tt @ TokenTree::Literal(_) => {
-            new_stream = quote!(text(#tt));
-        }
-        TokenTree::Punct(p) => emit_error!(p.span(), "'{}' not valid in item position", p),
+struct MoxExpr {
+    expr: syn::Expr,
+}
+
+impl Parse for MoxItem {
+    fn parse(input: syn::parse::ParseStream) -> syn::Result<Self> {
+        let parse_config = syn_rsx::ParserConfig::new()
+            .number_of_top_level_nodes(1)
+            .type_of_top_level_nodes(syn_rsx::NodeType::Element);
+        let parser = syn_rsx::Parser::new(parse_config);
+        let node = parser.parse(input)?.remove(0);
+
+        MoxItem::try_from(node)
     }
-    new_stream
 }
 
-impl ToTokens for MoxItem {
-    fn to_tokens(&self, tokens: &mut TokenStream) {
-        match self {
-            MoxItem::Tag(tag) => tag.to_tokens(tokens),
-            MoxItem::TagNoChildren(tag) => tag.to_tokens(tokens),
-            MoxItem::Content { stream, .. } => stream.to_tokens(tokens),
-            MoxItem::Fragment(_children) => todo!(),
+impl TryFrom<syn_rsx::Node> for MoxItem {
+    type Error = syn::parse::Error;
+
+    fn try_from(node: syn_rsx::Node) -> syn::Result<Self> {
+        use syn_rsx::NodeType::*;
+        match node.node_type {
+            Element => MoxTag::try_from(node).map(MoxItem::Tag),
+            Attribute => Err(node_parse_error(&node, "MoxItem")),
+            Text | Block => MoxExpr::try_from(node).map(MoxItem::Expr),
         }
     }
 }
 
-struct MoxTag {
-    span: Span,
-    name: Ident,
-    attributes: Vec<MoxAttr>,
-    children: Vec<MoxItem>,
+impl TryFrom<syn_rsx::Node> for MoxTag {
+    type Error = syn::parse::Error;
+
+    fn try_from(mut node: syn_rsx::Node) -> syn::Result<Self> {
+        use syn_rsx::NodeType::*;
+        match node.node_type {
+            Element => Ok({
+                let attributes: syn::Result<Vec<MoxAttr>> =
```

Comment:
If you remove the `syn::Result<>` here, can the question mark go on the next line instead of introducing another binding?

Reply:
I did `.collect::<syn::Result<Vec<_>>>()` instead. Type annotations are unavoidable.